explainer.md

```diff
@@ -251,6 +251,67 @@ function drawScene(view) {
 }
 ```
 
+### Enabling device features
+
+The use of some XR device features may introduce user privacy concerns, and as a result must gain the users consent before they are exposed to the page. In order to make use of these features an XRSession must explicitly request them by passing the `desiredFeatures` session creation option into a `requestSession` call.
+
+The `desiredFeatures` session creation option takes an array of strings. The `requestSession` call will iterate through the array and map each string to an `XRSessionFeature` enum. Items from the array that do not have a matching entry in the `XRSessionFeature` enum are ignored. (This is to ensure that newly added features do not break older UAs that don't recognize them.) Similarly, if the feature is not supported by the UA or XR device it is ignored. If the UA requires user consent for any of the supported features in the array, it must request that consent prior to the `requestSession` call resolving. Though UAs may have differing opinions on which features require explicit user consent, developers should expect that any entries in the `desiredFeatures` array will trigger a consent request visible to the user.
+
+An `XRSession` should still be returned even if the user denies consent for the desired features, and the feature's availability should be detected in a feature-specific manner once the session has been created. No differentiation is made at runtime between features that have not been requested, features that are not supported, and features for which consent has been denied.
+
+This samples shows how a developer would request a theoretical feature (here called `feature-name`) that exposes new data on the `XRFrame` and test for it's availability once the session has been created:
+
+```js
+// TODO: Make the example feature not theoretical.
+navigator.xr.requestSession({
+    mode: 'immersive-vr',
+    desiredFeatures: ['feature-name']
```

Comment:
I've been suggesting "geoAligned" as a feature.  Perhaps that would be a reasonable example?  It matches the pattern almost perfectly.